src/core/meshcop/border_agent_txt_data.cpp

```diff
@@ -152,6 +155,13 @@ Error TxtData::Prepare(uint8_t *aBuffer, uint16_t aBufferSize, uint16_t &aLength
     }
 #endif
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE && OPENTHREAD_CONFIG_BORDER_ROUTING_MULTI_AIL_DETECTION_ENABLE
+    if (Get<BorderRouter::MultiAilDetector>().IsDetected())
+    {
+        SuccessOrExit(error = encoder.AppendBooleanEntry(KeyExperimental::kMultiAil));
+    }
```

Comment:
I would go for this. So the key is added when "multi-ail-detector" is enabled and then include the status as  value of the key (`1` or `0`).
```suggestion
    if (Get<BorderRouter::MultiAilDetector>().IsEnabled())
    {
        SuccessOrExit(error = encoder.AppendEntry<uint8_t>(Key::kMultiAil, Get<BorderRouter::MultiAilDetector>().IsDetected() ? 1 : 0));
    }
```